File: RouterLib/src/main/java/com/kronos/router/interceptor/Interceptor.java
package com.kronos.router.interceptor;

import android.content.Context;
import android.os.Bundle;

import com.kronos.router.exception.RouteNotFoundException;
import com.kronos.router.model.HostParams;
import com.kronos.router.model.RouterParams;

import java.util.Map;

public interface Interceptor {
    void intercept(Chain chain) throws RouteNotFoundException;

    interface Chain {

        String url();

        Context getContext();

        Bundle getBundle();

        Map<String, HostParams> getHostParams();

        void proceed() throws RouteNotFoundException;

    }
}

File: RouterLib/src/main/java/com/kronos/router/interceptor/Interceptor.kt
package com.kronos.router.interceptor

import android.content.Context
import android.os.Bundle
import com.kronos.router.KRequest
import com.kronos.router.exception.RouteNotFoundException
import com.kronos.router.model.HostParams

interface Interceptor {
    @Throws(RouteNotFoundException::class)
    fun intercept(chain: Chain)

    interface Chain {

        val url: KRequest

        val context: Context

        val hostParams: Map<String, HostParams>?

        @Throws(RouteNotFoundException::class)
        fun proceed()
    }
}